### PR TITLE
Added icons and API for indeterminate checkmarks for the Tree class.

### PR DESCRIPTION
**Pull request by Phischermen**
This PR is a simpler version of #42682 that only includes icons and the ability to mark Tree items as indeterminate.

### CODE REVIEW

File: scene/gui/tree.cpp [hunk 1]
```diff
@@ -167,6 +167,17 @@ TreeItem::TreeCellMode TreeItem::get_cell_mode(int p_column) const {
...
 	cells.write[p_column].checked = p_checked;
+	cells.write[p_column].indeterminate = false;
+	_changed_notify(p_column);
+}
+
+void TreeItem::set_indeterminate(int p_column, bool p_indeterminate) {
+	ERR_FAIL_INDEX(p_column, cells.size());
+	// Prevent uncheck if indeterminate set to false twice
+	if (p_indeterminate == cells[p_column].indeterminate)
+		return;
```
Comment (KoBeWi) [inline scene/gui/tree.cpp:178]: Wrap the code under condition inside `{ }`, as per style guide.

File: scene/gui/tree.cpp [hunk 2]
```diff
@@ -1721,13 +1740,18 @@ int Tree::draw_item(const Point2i &p_pos, const Point2 &p_draw_ofs, const Size2
...
-						checked->draw(ci, check_ofs);
+					if (p_item->cells[i].indeterminate) {
+						indeterminate->draw(ci, check_ofs);
 					} else {
-						unchecked->draw(ci, check_ofs);
+						if (p_item->cells[i].checked) {
+							checked->draw(ci, check_ofs);
+						} else {
+							unchecked->draw(ci, check_ofs);
+						}
```
Comment (KoBeWi) [inline scene/gui/tree.cpp:1754]: Why is this nested? You could just use `else if`.

**Review by KoBeWi — APPROVED** [review]
Looks ok. We should add it in the editor where it makes sense, but could be done in another PR.

**Comment by akien-mga** [comment]
Needs a rebase, otherwise seems good to me.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**